File: targets/java-web/src/main/java/com/example/vulnapp/vulns/Vuln22Encrypt.java
package com.example.vulnapp.vulns;

import com.example.vulnapp.vulns.crypto.AesCipher;
import org.springframework.web.bind.annotation.*;

@RestController
public class Vuln22Encrypt {
    @GetMapping("/vuln22/encrypt")
    public String encrypt(@RequestParam String data) {
        // hard-coded key (VULN-23) + static IV (VULN-22) sinks live in AesCipher
        return AesCipher.encrypt(data);
    }
}

File: targets/kotlin-web/src/main/kotlin/com/example/vulnapp/vulns/Vuln22Encrypt.kt
package com.example.vulnapp.vulns

import org.springframework.web.bind.annotation.*


@RestController
class Vuln22Encrypt {

    @GetMapping("/vuln22/encrypt")
    fun encrypt(@RequestParam data: String): String = com.example.vulnapp.vulns.crypto.AesCipher.encrypt(data)
}
